toolkit/tools/downloader/downloader.go

```diff
@@ -34,14 +34,17 @@ var (
 	tlsClientCert = app.Flag("certificate", "TLS client certificate to use when downloading files.").String()
 	tlsClientKey  = app.Flag("private-key", "TLS client key to use when downloading files.").String()
 
-	dstFile   = app.Flag("output-file", "Destination file to download to").Short('O').String()
-	prefixDir = app.Flag("directory-prefix", "Directory to download to").Short('P').String()
-	srcUrl    = app.Arg("url", "URL to download").Required().String()
+	dstFile     = app.Flag("output-file", "Destination file to download to").Short('O').String()
+	prefixDir   = app.Flag("directory-prefix", "Directory to download to").Short('P').String()
+	rpmFilename = app.Arg("rpm-filename", "RPM to download").Required().String()
+	srcUrls     = app.Arg("url-list", "URLs to download").Required().Strings()
```

Comment:
This tool is mean to roughly mimic `wget`, I'd like to keep the basic behavior roughly the same.

Thoughts on having `--url-list` require fully formed URLs and drop `--rpm-filename`? That would keep the behaviour (we try each URL in the list in order, failing if they all fail).

Then we could just generate the URL list in make.

Reply:
Move that URL generating work to make is doable. Does the downloader have the potential to be used elsewhere? I can imagine if the trailing part is completely different for the URLs in other scenarios, dropping `--rpm-filename` makes it more generic.

Reply:
+1 to a simple change where `--url-list` can have more than one entry. This way the tool is still a generic downloader (like @dmcilvaney said) but we have a way to ask it to download multiple files at once.

Having said that, now we have to deal with how we handle new scenarios:
1. Download failures, where we can:
    - fail on the first unsuccessful download,
    - fail when there was at least one unsuccessful download but try to download all files before exiting,
    - never fail but display a warning.
2. Downloading same-named file from separate sources: how do we save them, so they don't override each other?

I understand the current change is to address the issue of downloading one file but from a set of X potential sources. That's just not what this tool is for, however - it's meant to accept a URL and try to download the file using it. We could try now to extend the use cases it's suppose to handle but I think that'll unnecessarily complicate the implementation, so my suggestion would be to handle that on the caller's side.

Reply:
The current changes reflect the original behavior from `toolchain.mk` and `downloader.go` combined, that is, mark a success if it downloads the file from a URL, skip the remaining URLs and move on to the next RPM file, or mark a failure if it is unable to download the file from any of the given URLs and move on to the next RPM file.

@dmcilvaney and I proposed handling the bug from the caller's side exclusively as one of the possible routes but I liked the idea of moving the for loop into the downloader by @dmcilvaney. IMHO with `--url-list` being dropped it stays generic and becomes compatible with more use cases at the cost of internal complexity. The positive side is it reduces the complexity of the Makefile which is harder to read and more error-prone than Golang, which is why I chose this route.

In order to reduce the complexity on the Golang side, maybe we can try to keep the original downloader as much as possible in a helper function.